src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,203 @@
+ */
+public class Parser {
+    protected enum CommandEnum {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND
+    }
+
+    /**
+     * Instantiates a new ToDo task and returns it.
+     * @param splitInputs An array containing the input by the user.
+     * @return The new ToDo task.
+     * @throws IncompleteDescException If name of the task is not given.
+     */
+    private static ToDo getTodo(String[] splitInputs) throws IncompleteDescException {
+        if (splitInputs.length <= 1 || splitInputs[1].isBlank()) {
+            throw new IncompleteDescException(
+                    "The description of a todo cannot be empty!");
+        }
+        return new ToDo(splitInputs[1]);
+    }
+
+    /**
+     * Parses the given string into a LocalDate object.
+     *
+     * @param date The given string representation of the date to be parsed.
+     * @return The parsed LocalDate.
+     */
+    public static LocalDate parseDate(String date) {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+        return LocalDate.parse(date, dtf);
+    }
+
+
+    /**
+     * Instantiates a new Deadline object and returns it.
+     * @param splitInputs An array containing the input by the user.
+     * @return A new Deadline object.
+     * @throws IncompleteDescException If name or due date/time or both is/are not provided.
+     * @throws InvalidInputException If due date/time is given in the wrong format.
+     */
+    private static Deadline getDeadline(String[] splitInputs) throws IncompleteDescException, InvalidInputException {
+        if (splitInputs.length <= 1) {
+            throw new IncompleteDescException(
+                    "The description of a deadline cannot be empty!");
+        }
+        int endIndex = splitInputs[1].indexOf("/by");
+        if (endIndex < 0) {
+            throw new IncompleteDescException(
+                    "Please add the due date/time!");
+        }
+        String name = splitInputs[1].substring(0, endIndex).strip();
+        String end = splitInputs[1].substring((endIndex + 3)).strip();
+        if (name.isBlank()) {
+            throw new IncompleteDescException(
+                    "The description of a deadline cannot be empty!");
+        }
+        if (end.isBlank()) {
+            throw new IncompleteDescException(
+                    "Please add the due date/time!");
+        }
+        try {
+            LocalDate endLd = Parser.parseDate(end);
+            end = Formatter.formatDateForPrint(endLd);
+            if (endLd.isBefore(LocalDate.now())) {
+                throw new InvalidInputException(
+                        "The given deadline (yyyy/mm/dd) " + end + " has passed.");
+            }
+            return new Deadline(name, endLd);
+        } catch (DateTimeParseException e) {
+            throw new InvalidInputException(
+                    "Please enter the date/time in \"yyyy/mm/dd\" format.");
+        }
```

Comment:
Do you think dividing logical units into separate blocks would improve readability? 🤔